app/models/project.rb

```diff
@@ -1,4 +1,19 @@
 class Project < ApplicationRecord
   validates_presence_of :name, :material
   belongs_to :challenge
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def count_contestants
+    Contestant.joins(:projects).where(projects: {id: self.id}).count
+  end
+
+  def average_contestant_experience
+    years = Contestant.joins(:projects).where(projects: {id: self.id}).average(:years_of_experience)
```

Comment:
Cool - also good use of `joins` here, but again it's made a lot harder than is needed, since we can also use the `contestants` association to get all the related contestants to this one Project. 
You can still call `.average(:years_of_experience)` as you did, to get that calculation from the database.